stockroom/diff.py

```diff
@@ -0,0 +1,126 @@
+from typing import NamedTuple
+
+from hangar import Repository
+from rich.jupyter import print
+from rich.table import Table
+from rich.segment import Segment
+from rich.console import Console
+
+from .parser import tagkey, SEP
+
+class ParsedDiff(NamedTuple):
+    tagName: str
+    ref_sample: str = ''
+    dest_samples: list = list()
+
+class TagDiff:
+    """
+    The Diff object for tags
+    """
+
+    def __init__(self, ref_commit: str = None, dest_commit: str = None,
+                    root: str = None):
+        self.ref_commit = ref_commit
+        self.dest_commit = dest_commit
+        self.repo = Repository(root)
+        self.all_keys = set()
+
+        self._added: dict = dict()
+        self._deleted: dict = dict()
+        self._mutated: dict = dict()
+
+        self.parse_diff()
+
+    def gen_tables(self):
+    """
+    Generate the Rich Table using parsed data.
+    """
+
+        table = Table(title='Diff %s..%s'%(self.ref_commit[2:7], self.dest_commit[2:7]))
+
+        # Generate Columns
+        column_headings = ['Tags'] + [self.ref_commit, self.dest_commit]
+        for column in column_headings:
+            table.add_column(column[:10], no_wrap=True)
+
+        # Generate rows
+        for row in self.all_keys:
+            ref_row = ""
+            dest_row = ""
+
+            if row in self._added:
+                dest_row = "[green] %s [/green]"%(self._added[row])
```

Comment:
Use fstrings whenever possible. Fstrings is the new feature added to 3.6 and since we support only 3.6+, it's preferable to use that

Reply:
will do that